system/lib/wasmfs/file.cpp

```diff
@@ -178,12 +178,13 @@ int Directory::Handle::insertMove(const std::string& name,
       it->second.file->locked().setParent(nullptr);
       it->second = entry;
     }
-    file->locked().setParent(getDir());
   } else {
     // This backend doesn't use the dcache.
     assert(getDir()->maintainsFileIdentity());
   }
 
+  file->locked().setParent(getDir());
```

Comment:
Is this a separate change?

Reply:
Somewhat, but it's a fix for a bug that @mere-human noticed [here](https://github.com/emscripten-core/emscripten/pull/18248#discussion_r1032263406) as part of this PR, and this PR also adds a test that requires this change to pass.